### PR TITLE
Added support for TCP lifecycle check when Ship ip in IPv6 format. (Fixes #183)

### LINKED ISSUE #183: Lifecycle check is not working if the Ship ip is in IPv6 format
TcpPortPinger class in lifecycle.py always expects the ship ip to be in IPv4 format and if the IP is in IPv6 format the lifecycle check fails [lifecycle.py](https://github.com/signalfx/maestro-ng/blob/c2babac1640169236b3d3bee1ccc7b3405ce2f0c/maestro/lifecycle.py#L68)

```
   def _test(self, container=None):
        try:
            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            s.settimeout(1)
            s.connect((self.host, self.port))
            s.close()
            return True
        except Exception:
            return False
```

We should first check if the IP is in IPv6 or IPv4 format and then updated the socket.AF.INET function to socket.AF.INET6 if its an IPv6 format. I will try to update the code and raise a Pull request to update the same. If you have any suggestion please let us know.

Thanks,
Rahul Jain

### PR DESCRIPTION
**Pull request by zsuzhengdu**

### CODE REVIEW

File: maestro/lifecycle.py [hunk 1]
```diff
@@ -65,10 +65,7 @@ def __repr__(self):
 
     def _test(self, container=None):
         try:
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(1)
-            s.connect((self.host, self.port))
-            s.close()
+            create_connection((self.host, self.port), 1)
```
Comment (mpetazzoni) [inline maestro/lifecycle.py]: I prefer to keep `socket.create_connection` to make it obvious that this comes from the `socket` module and is not some local method.

**Comment by zsuzhengdu** [comment]
Done, as per suggestion.

File: maestro/lifecycle.py [hunk 2]
```diff
@@ -65,10 +65,7 @@ def __repr__(self):
 
     def _test(self, container=None):
         try:
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(1)
-            s.connect((self.host, self.port))
-            s.close()
+            socket.create_connection((self.host, self.port), 1)
```
Comment (mpetazzoni) [inline maestro/lifecycle.py]: You still need to close the connection, don't you?

Comment (zsuzhengdu) [inline maestro/lifecycle.py]: I think you are right; close socket is needed on successful connection (https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/socket.py#L714). 

How about
```
s = socket.create_connection((self.host, self.port), 1)
s.close()
```

**Comment by mpetazzoni** [comment]
Thanks!

**Pull request merged**